### DISCUSSION TITLE
Why is it taking bootstrap? and not my code?

### DISCUSSION THREAD

**Discussion by Elindo586** [vercel/next.js#36786] [Help] [answered]
At times my code works, at times, I have to use bootstrap.

My head is about to explode.

This works perfect on one page:

Page:
```
<div id="oilanalysis2">
            <Image
              src="/images/oil-analysis2.png"
              alt="oil analysis two pic"
              width={700}
              height={400}
            />
</div>
```
From global CSS:

```
#oilanalysis2 {
  padding-top: 1em;
  padding-left: 1em;
  padding-right: 2em;
  padding-bottom: 1em;
  height: auto;
}
```

^^^^ If that works above....  why wouldn't this work below??

From page:

  ```
<div id="fixedpic">
              <Image
                src="/images/radial-piston-pump.png"
                alt="fixed-piston-pump"
                width={168}
                height={130}
              />
            </div>
```

From global CSS:
```

#fixedpic {
  display: block;
  margin-left: auto;
  margin-right: auto;
}
```

^^^ This code above worked on my React-App... I just wrapped it in a <div> to work with Next.js..... but at times it seems to take the code, and other times it doesn't take the code.

What exactly is going on? 

.... BTW....  I am fixing it by adding a bootstrap class "d-flex justify-content-center",  but not understanding how its working is not a good feeling and will get me in time delays..

**Comment by icyJoseph** [comment, marked as answer]
Hi,

Perhaps if you have a repository I could most definitely point at the what's going on. You can invite me as collaborator, and later on delete me, I can comment back here.

At the top of my head, you are trying to win a specificity war against inline styles, which have higher specificity than id selector and class selectors.

The inline styles come from `next/image` best practices, and the properties set by the inline style CSS declaration override those of id selectors and class selectors, which is why it looks like some rules work, and some do not, it just depends on what CSS declarations the rule contains.

The Bootstrap class you are using, going by the name, is using flex to center, not margin auto, so there's nothing in the inline style that could override it.

**Reply by Elindo586** [reply]
Thanks a lot...

I will add the project to a repository.

Say... what would be the proper way to override/change the inline style if needs to be?  ... or what would be the best way to work it to modify it into the results you are looking for?

**Reply by icyJoseph** [reply]
You don't need to style the image directly. Wrap it on a div/span, and style that. Of course this is for positioning, and aligning it with a layout you might have. 

> Notice that the next/image component renders more than just an image.

If you want to do objectFit, there's a prop for that, or if you want to make the image rendering pixelated and such, you can do that via class names.

The layout modes on `next/image` are what you should use to make the image behave well within your wrapper. I think, you should use `layout="fill"` as of now.

Maybe this article explains it better than me, https://www.peterlunch.com/snippets/next-image-styling. Please do note that, nowadays, `next/image` uses a `span` as a wrapper.

At the end of the day this is a CSS process, you may remember that `!important`, overrides all, but getting into `!important` battles, is a lost cause.

**Reply by icyJoseph** [reply]
Did you manage to sort this out?

**Reply by Elindo586** [reply]
Yes... was looking to mark your last post as an answer.
